src/mongocrypt-ctx-datakey.c

```diff
@@ -471,17 +471,24 @@ mongocrypt_ctx_datakey_init (mongocrypt_ctx_t *ctx)
    ctx->vtable.cleanup = _cleanup;
 
```

Comment:
I suggest adding a `key_material` field to the `_mongocrypt_ctx_opts_spec_t` struct. Set it to `OPT_OPTIONAL` in `mongocrypt_ctx_datakey_init`.

That should prevent callers from accidentally using `mongocrypt_ctx_setopt_key_material` on the wrong context. E.g. if the suggestion is implemented, a call to `mongocrypt_ctx_setopt_key_material` then `mongocrypt_ctx_encrypt_init` would result in an error.

Reply:
Neat. Was not aware of this safety check. Will implement as suggested.